lib/src/http/utils.dart

```diff
@@ -46,22 +48,77 @@ Duration calculateBackOff(RequestAutoRetry autoRetry) {
   return backOff;
 }
 
-Duration _calculateExponentialBackOff(RequestAutoRetry autoRetry) {
+Duration _calculateExponentialBackOff(RequestAutoRetry autoRetry,
+    {@visibleForTesting Random random}) {
+  if (autoRetry.backOff.withJitter ?? false) {
+    final jitteredBackOff = _calculateAdvancedExponentialJitteredBackOffInMs(
+        autoRetry,
+        random: random);
+    // If we're over the maximum duration, fall back to a fixed maxInterval with full jitter
+    if (jitteredBackOff > autoRetry.backOff.maxInterval.inMilliseconds) {
```

Comment:
This essentially falls back to what the "Full jitter" approach was doing when it hit the max interval. The Polly jitter algorithm doesn't have a notion of a max time, just a max number of retries 🤷‍♂